Change desktop entry

Grafcube opened the pull request:
Changed Exec to /usr/bin and Name[de_DE].

Murmele commented:
In the flatpak package the executable is not under the /usr/bin. So it may not work anymore

Grafcube commented:
I don't know much about how flatpak packaging works. The PKGBUILD in my PR #16 symlinks the executable to `/usr/bin/gittyup` so the desktop entry also needs to point there.

As far as I know, launching an application via flatpak needs a different command anyway so the `Exec` value needs to be different. There could be 2 different files for this, which could be useful if you start putting pre-built binaries in releases for non-flatpak installs. Alternatively I could make the PKGBUILD apply a patch in prepare that changes the necessary exec but this will limit it to the arch package.

The name change does not cause any issues I think.

Murmele commented:
I find it cleaner to have it mainlined so lets find a way to fix this

Grafcube commented:
I built the flatpak from this branch and it worked correctly. The Exec line in the desktop entry for the flatpak version is `/usr/bin/flatpak run ...` so I think flatpak either generates its own desktop entry or it modifies the existing one. Either way I think it's safe to merge this and keep only one entry.

Murmele commented:
I will try out and then I will merge

Murmele commented:
@Grafcube For me it does not work. I checked the compiled flatpak manifest and there was nothing changed. If I remember me correctly there is a command for the flatpak manifest to update the manifest. I will check it out

exactly-one-kas commented:
This is probably related to #111
We'll need a separate desktop file for Flatpak and for global installation.

Maybe use CMake's `configure_file` to generate both files from a template?

alerque commented on rsrc/linux/com.github.Murmele.Gittyup.desktop:
```diff
@@ -2,11 +2,11 @@
 Name=Gittyup
 Comment=Graphical Git client for Windows, Linux and macOS
-Exec=Gittyup
+Exec=/usr/bin/gittyup
```
Why specify the full path here? For distros that install `*.desktop` files it is almost always safe to assume they have the bin files in a proper location in the `$PATH`. Only in the event of a totally non-standard location (such as `/opt/blah/...` would a full path be appropriate here. In case they install to a location outside `$PATH` it would be up to the distro or installer to set this, not for an upstream project to predict.

```suggestion
Exec=gittyup
```

Grafcube replied:
Thanks for the change.

alerque commented on rsrc/linux/com.github.Murmele.Gittyup.desktop:
```diff
@@ -2,11 +2,11 @@
 Categories=Development;
 X-Desktop-File-Install-Version=0.1
-Name[de_DE]=com.github.Murmele.Gittyup.desktop
+Name[de_DE]=Gittyup
```
```suggestion
```

Localized names only makes sense if the localization is different from the default, specifying a German `Name[de_DE]` that is just the same as `Name` doesn't make any sense. All other locales will just be falling back to the default anyway, why special case that one?

alerque commented on rsrc/linux/com.github.Murmele.Gittyup.desktop:
```diff
@@ -2,11 +2,11 @@
 Terminal=false
 Type=Application
 Categories=Development;
 X-Desktop-File-Install-Version=0.1
```
What is this key supposed to do? And is in really necessary? From what I know of the desktop file spec I would have just done this instead:

```suggestion
Version=1.0
```

If there is some good reason for the X-... key and version info that's fine, but I suspect that was just copied from somewhere and it actually doesn't make any sense here.

alerque reviewed (approved):
LGTM now.

Murmele commented:
@Grafcube @alerque is this still needed?

Grafcube commented:
Probably not. I switched to the flatpak long ago but I originally made this PR to simplify making the PKGBUILD a little. But someone else ended up making the PKGBUILD so I dropped it. I don't really have anything to do with this PR anymore so it would probably only be of interest to packagers (since it looks like flatpak doesn't even use this file). I think it's safe to merge once I resolve the conflicts. After that it's up to the package maintainers to use it.

alerque reviewed (approved):
Yes this is still relevant and needed.

Murmele commented:
Not needed anymore after #397 is merged
